app/src/main/java/org/coepi/android/ui/symptoms/SymptomsViewModel.kt

```diff
@@ -63,7 +63,17 @@ class SymptomsViewModel(
         disposables += checkedSymptomTrigger
             .withLatestFrom(selectedSymptomIds)
             .subscribe { (selectedSymptom, selectedIds) ->
-                selectedSymptomIds.onNext(selectedIds.toggle(selectedSymptom.symptom.id))
+                if (selectedSymptom.symptom.id == SymptomId.NONE && !selectedSymptom.isChecked) {
+                    // "No symptoms" selected. Unselect all other symptoms that were selected/highlighted
+                    selectedSymptomIds.onNext(emptySet<SymptomId>().toggle(selectedSymptom.symptom.id))
+                } else if (selectedSymptom.symptom.id == SymptomId.NONE && selectedIds.size == 1) {
+                    // "No symptoms" selected with no other symptoms highlighted. Only toggles NONE
+                    selectedSymptomIds.onNext(selectedIds.toggle(selectedSymptom.symptom.id))
+                } else {
+                    selectedSymptomIds.onNext(
+                        selectedIds.minus(SymptomId.NONE).toggle(selectedSymptom.symptom.id)
+                    )
+                }
```

Comment:
I think this is good and it's working, but we can optimize a little for readability.

We could model it as a flow: First toggle "normally", and then solve the conflicts. Something like:

```kotlin
val newSelectedIds = selectedIds.toggle(selectedSymptom.symptom.id)
val resolvedSelectedIds = newSelectedIds.solveConflicts(...)
```
If you put `solveConflicts(...)` in a `Set` extension, it improves another bit:
```kotlin
selectedIds.toggle(selectedSymptom.symptom.id).solveConflicts(...)
```

Lastly, using `when` instead of `if`/`else` improves it yet another bit.

We'd end with something along the lines of this:

```kotlin
                selectedSymptomIds.onNext(selectedIds.toggle(selectedSymptom.symptom.id) .solveConflictingSelections(selectedSymptom.symptom.id,
                        !selectedSymptom.isChecked)
                )

private fun Set<SymptomId>.solveConflicts(selectedId: SymptomId, wasChecked: Boolean): Set<SymptomId> =
    when {
        !wasChecked -> this // We can't run in conflicts when unchecking, so return set unchanged
        selectedId == NONE -> setOf(selectedId) // Selected NONE: deselect everything else
        contains(NONE) -> minus(NONE) // Selected something other than NONE and NONE is selected: deselect NONE
        else -> this
    }
```

Reply:
P.S. Sorry for the late (post-merge) review! I'll try to be quicker next time 🙂

Reply:
@i-schuetz thanks for the suggestion! That definitely seems like a cleaner implementation. I will create a new PR with these changes added in soon.